### DISCUSSION TITLE
karmaTestExplorer.customLauncher,"base"

### DISCUSSION THREAD

**Discussion by PandeleFlorin** [lucono/karma-test-explorer#80] [Issues / Help]
Hello
What is the list with accepted values for this setting?
I have to use firefox headless!!

It says "browsers" in the doc but it ignores setting it to "Firefox" or "Firefoxheadless" and Karma server complained initially:

```
11 07 2023 15:34:09.421:ERROR [launcher]: Cannot load browser "KarmaTestExplorer_CustomLauncher"!
  Error: No provider for "launcher:Chrome". Can not use provider from the parent!
```
now it says:
```
12 07 2023 14:52:11.354:ERROR [launcher]: Cannot load browser "KarmaTestExplorer_CustomLauncher"!
  Error: No provider for "launcher:firefox". Can not use provider from the parent!

```
Tried setting VS code, workspace setting:
```
{
  "karmaTestExplorer.projectType": "angular",
  "karmaTestExplorer.testFramework": "jasmine",
  "karmaTestExplorer.projectWorkspaces":
  [

    {
      "rootPath": ".",
      "karmaConfFilePath": "angular/karma.conf.js",
      "testFiles": ["angular/src/app/angular/**/*.spec.ts"],     
    }
],
"karmaTestExplorer.customLauncher": {

  "base": "Firefox",
  "flags": [
    "--headless",
    "--disable-gpu",
    "--disable-dev-shm-usage",
    "--remote-debugging-port=9222"
  ]
}
}
```

**Comment by lucono** [comment]
Any value that works when used in the `browsers` field of your Karma config file when testing your project directly with Karma on the command line, should also work in the `karmaTestExplorer.browser` setting of Karma Text Explorer. This also means that the relevant dependencies must be installed.

In the case of firefox, the firefox browser must be installed on the machine on which you are running VSCode, and the [`karma-firefox-launcher`](https://www.npmjs.com/package/karma-firefox-launcher) dependency must be installed in your project.

**Reply by PandeleFlorin** [reply]
Well I tried that, karma tests do work on command prompt but not form the plugin. Somehow Karma refuses to load the launcher, because "Can not use provider from the parent"...whatever that is supposed to mean. Any ideas?

**Reply by lucono** [reply]
Are you able to reproduce the issue with a skeletal or newly generated Angular project, where specifying the `browsers` setting as above gives the same error?

Please open a new [bug report](https://github.com/lucono/karma-test-explorer/issues/new?template=bug_report.md) with all the necessary details, including the minimal reproduction project. That will help identify and address the issue.
